dev/config/apollo-federation/supergraph.graphql

```diff
@@ -1587,6 +1604,15 @@ type Query
   authorization: Authorization! @join__field(graph: PUBLIC)
   btcPriceList(range: PriceGraphRange!): [PricePoint] @join__field(graph: PUBLIC)
   businessMapMarkers: [MapMarker!]! @join__field(graph: PUBLIC)
+
+  """Returns an estimated conversion rate for the given amount and currency"""
+  currencyConversionEstimate(
+    """Amount in major unit."""
+    amount: Float!
+
+    """Currency of the amount to be converted."""
+    currency: DisplayCurrency = "USD"
+  ): CurrencyConversionEstimate! @join__field(graph: PUBLIC)
```

Comment:
the wording make me thing this should be a mutation (ie: it's an action)

Reply:
how is that different from fetching the price directly?

Reply:
> how is that different from fetching the price directly?

the purpose is to reduce queries and overhead on price calculation... even for us is difficult to follow and do a proper implementation, for example please check galoy pay implementation https://github.com/GaloyMoney/galoy/blob/main/apps/pay/lib/use-realtime-price.tsx#L85 (or search `** offset`) we did not implement it correctly. 

In the case of ecommerce plugins we need an easy way to convert fiat prices (in major unit) to wallet currency (minor unit) so this query solve 2 things: reduces the amount of requests to create an invoice from 3 (list currencies, realtime, create invoice) to 2 and reduce the friction for developers in terms of proper calculation using base, offset and fractionDigits

Reply:
> the wording make me thing this should be a mutation (ie: it's an action)

uhmm for me it is a query because it does not modify anything on the server side... @vindard @bodymindarts wdyt? 

@nicolasburtey I also have doubts about the name, so please let me know if you prefer something else

Reply:
> > the wording make me thing this should be a mutation (ie: it's an action)
> 
> uhmm for me it is a query because it does not modify anything on the server side... @vindard @bodymindarts wdyt?
> 
> @nicolasburtey I also have doubts about the name, so please let me know if you prefer something else

yup I read back https://graphql.org/learn/queries/ and you are right I think, there is no side effect to be expected here. but the cue on using a verb can generally be a good guess (https://github.com/GaloyMoney/galoy/issues/3699)

Reply:
how about:
```
walletCurrencyAmount { 
    amount: Float!
    currency: DisplayCurrency
}
```

return
```
type WalletCurrencyAmount
  @join__type(graph: PUBLIC)
{
  btcSatAmount: SatAmount!
  usdCentAmount: CentAmount!

  currency: Currency!
  id: ID!
  timestamp: Timestamp!
}
```

Reply:
Updated according to ACE meeting https://github.com/GaloyMoney/galoy/pull/4222/commits/bc722bc1d5c06b0a98c9c32669b084091f774c82